inline-r/src/Language/R/QQ.hs

```diff
@@ -118,199 +90,63 @@ parse txt = runIO $ do
-                 unhexpIO . Expr n . vector
-             |]
-      (returnIO . hexp -> iot) ->
-        [| unhexpIO =<< iot |]
-
-instance TH.Lift (IO (HExp s a)) where
-  lift = runIO >=> \case
-    Nil -> [| return Nil |]
-    Symbol (returnIO -> x0io) (returnIO -> x1io) (returnIO -> x2io) ->
-      [| R.withProtected x0io $ \x0 ->
-         R.withProtected x1io $ \x1 ->
-           fmap (Symbol x0 x1) x2io
-        |]
-    List (returnIO -> x0io) (returnIO -> x1io) (returnIO -> x2io) ->
-      [| R.withProtected x0io $ \x0 ->
-         R.withProtected x1io $ \x1 ->
-           fmap (List x0 x1) x2io
-        |]
-    Env (returnIO -> x0io) (returnIO -> x1io) (returnIO -> x2io) ->
-      [| R.withProtected x0io $ \x0 ->
-         R.withProtected x1io $ \x1 ->
-           fmap (Env x0 x1) x2io
-        |]
-    Closure (returnIO -> x0io) (returnIO -> x1io) (returnIO -> x2io) ->
-      [| R.withProtected x0io $ \x0 ->
-         R.withProtected x1io $ \x1 ->
-           fmap (Closure x0 x1) x2io
-        |]
-    Promise (returnIO -> x0io) (returnIO -> x1io) (returnIO -> x2io) ->
-      [| R.withProtected x0io $ \x0 ->
-         R.withProtected x1io $ \x1 ->
-           fmap (Promise x0 x1) x2io
-        |]
-    Lang (returnIO -> x0io) (returnIO -> x1io) ->
-      [| R.withProtected x0io $ \x0 ->
-           fmap (Lang x0) x1io
-        |]
-    Special                  x0  -> [| return $ Special x0 |]
-    Builtin                  x0  -> [| return $ Builtin x0 |]
-    Char      (returnIO -> x0io) -> [| fmap Char      x0io |]
-    Logical   (returnIO -> x0io) -> [| fmap Logical   x0io |]
-    Int       (returnIO -> x0io) -> [| fmap Int       x0io |]
-    Real      (returnIO -> x0io) -> [| fmap Real      x0io |]
-    Complex   (returnIO -> x0io) -> [| fmap Complex   x0io |]
-    String    (returnIO -> x0io) -> [| fmap String    x0io |]
-    DotDotDot (returnIO -> x0io) -> [| fmap DotDotDot x0io |]
-    Vector x0 (returnIO -> x1io) -> [| fmap (Vector x0) x1io |]
-    Expr   x0 (returnIO -> x1io) -> [| fmap (Expr x0) x1io |]
-    Bytecode -> [| return Bytecode |]
-    ExtPtr _ _ _ -> violation "TH.Lift.lift HExp" "Attempted to lift an ExtPtr."
-    WeakRef (returnIO -> x0io) (returnIO -> x1io)
-            (returnIO -> x2io) (returnIO -> x3io) ->
-      [| R.withProtected x0io $ \x0 ->
-         R.withProtected x1io $ \x1 ->
-         R.withProtected x2io $ \x2 ->
-           fmap (WeakRef x0 x1 x2) x3io
-        |]
-    Raw (returnIO -> x0io) -> [| fmap Raw x0io |]
-    S4  (returnIO -> x0io) -> [| fmap S4  x0io |]
-
-unhexpIO :: HExp s a -> IO (SEXP s a)
-unhexpIO = unsafeRToIO . unhexp
+isAnti :: SEXP s 'R.Char -> Bool
+isAnti (hexp -> Char (Vector.toString -> name)) = "_hs" `isSuffixOf` name
+isAnti _ = error "Impossible"
+
+-- | Chop antiquotation variable names to get the corresponding Haskell variable name.
+chop :: String -> String
+chop name = take (length name - 3) name
```

Comment:
This is pre-existing function but the magic `3` has irked me for a while. Maybe something like

```
varSuffix = "_hs"
isAnti … = varSuffix `isSuffixOf` name
chop name = take (length name - length varSuffix) name
```

would be nicer. Or at least less out of thin air. Feel free to ignore this though.